lib/tasks/test.rake

```diff
@@ -2,16 +2,6 @@ require_relative './evm_test_helper'
 
 if defined?(RSpec)
 namespace :test do
-  task :initialize do
-    if ENV['RAILS_ENV'] && ENV["RAILS_ENV"] != "test"
-      warn "Warning: RAILS_ENV is currently set to '#{ENV["RAILS_ENV"]}'. Forcing to 'test' for this run."
-    end
-    ENV['RAILS_ENV'] = "test"
-    Rails.env = 'test' if defined?(Rails)
-
-    ENV['VERBOSE'] ||= "false"
-  end
```

Comment:
If dropping this and setting RAILS_ENV=test in the github actions workflow works, we'll need to deal with all of the plugins that call setup_db and spec_deps.

```
% rg -l -M 300 -e "spec_deps|setup_db" ./ `gem env home`
./lib/generators/manageiq/provider/provider_generator.rb
./lib/generators/manageiq/plugin/templates/lib/tasks_private/spec.rake
./lib/tasks/test_providers_common.rake
./lib/tasks/test.rake
./lib/tasks/test_vmdb.rake
/Users/joerafaniello/.gem/ruby/2.7.5/gems/acts_as_tree-2.9.1/test/acts_as_tree_test.rb
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-autosde-c6c89b573336/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-kubevirt-1ff1ab293095/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ovirt-dc299d714d78/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-automation_engine-2e4536f90465/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-cisco_intersight-415227fd0821/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-nsxt-bf25dd73b1ed/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-ui-classic-22d98918ab27/Rakefile
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-consumption-5e54c15961d0/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-oracle_cloud-63b6a6954762/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ibm_terraform-265f4ef4267e/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-amazon-546065432644/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ibm_power_vc-a187c2ce8883/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-redfish-53397a19a47e/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-lenovo-5fe03693e916/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-azure_stack-fc367ff913ad/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-schema-ed91fdbbf8a7/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ibm_cic-af476f32502b/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-azure-98e9bbf10c1d/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-kubernetes-60fbf960c707/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-openshift-47f4c894cb67/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-vmware-9dfcf0bea396/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-api-440b0d235272/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-nuage-bec9dd38b3fa/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ansible_tower-613453fcefc2/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-ui-classic-4daae98b9e6a/Rakefile
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-content-240cb8128e44/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-decorators-f85c804b1d51/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ibm_power_hmc-7be919dcd7b8/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-scvmm-498dbc9d4ab8/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-ibm_cloud-6b37b87cc92a/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-google-bc51464c1803/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-foreman-7adc6bb6fd14/lib/tasks_private/spec.rake
/Users/joerafaniello/.gem/ruby/2.7.5/bundler/gems/manageiq-providers-openstack-cde0ed4f8fd7/lib/tasks_private/spec.rake

Reply:
They depend on setting up the test database by dynamically setting the RAILS_ENV via the initialize rake task prerequisite task, so we'll need to set this variable in all of their test suites.